api/envoy/api/v2/endpoint/load_report.proto

```diff
@@ -105,6 +106,49 @@ message EndpointLoadMetricStats {
   double total_metric_value = 3;
 }
 
+// Stats per downstream host.
```

Comment:
More descriptive comment explaining the situation in which this is used to report load.

Reply:
Objective:
We want the backends of the service to report load stats per frontend it gets requests from.

Proposal (this PR):
Since the backend is reporting load stats per frontend it gets requests from , I propose to call this downstream endpoint stats.
 
E.g:
```
LoadStatsRequest {
ClusterStats {
cluster_name = 
...
DownstreamEndpointStats {
  address = <frontend-1-ip>
  total_successful_requests = 2
  total_requests_in_progress = 1
  total_error_requests = 0
}
DownstreamEndpointStats {
  address = <frontend-2-ip>
  total_successful_requests = 3
  total_requests_in_progress = 1
  total_error_requests = 0
}
...
}
}
```

Please do suggest any potential issues or better design ideas.

Reply:
How will each downstream be identified to the upstream?

Reply:
Upstream identifies each downstream by its IP address.

Reply:
OK, FYI, at least in the current code the downstream IP can be synthetic based on proxy proto, XFF, etc. I would think a bit about which downstream you want (the direct IP address? Logical IP address)? I think it would be helpful to add some more docs?

Reply:
@karthikbox I think it's fine to add in features like this to LRS that also make sense to gRPC-LB. While doing so, we need to think about how they might be used in a middle-box/edge/sidecar proxy scenario as well, to make sure they still make sense in Envoy proper.

To me, this is a way to allow Envoys to stream load information about the per-client breakdown of traffic to the management server. Today, we report just the cluster/locality distribution of traffic. Knowing where that traffic comes from might be also interesting, e.g. you might use the fact that 70% of your traffic on an Envoy is arriving from outside of your zone to adjust load assignments elsewhere in the mesh. Arguably this example is contrived and you would infer this from existing LRS reports.

Are there other examples of how you would use these client load reports in Envoy? Would you ever want to associate client traffic with selected cluster in any of them?